x-pack/plugins/apm/public/components/alerting/alerting_flyout/index.tsx

```diff
@@ -20,10 +20,28 @@ interface KibanaDeps {
 
 export function AlertingFlyout(props: Props) {
   const { addFlyoutVisible, setAddFlyoutVisibility, alertType } = props;
+  const { serviceName } = useParams<{ serviceName?: string }>();
   const {
     services: { triggersActionsUi },
   } = useKibana<KibanaDeps>();
 
+  const alertTypeName = alertType
+    ? ALERT_TYPES_CONFIG[alertType].name
+    : undefined;
+  const alertName = alertTypeName
+    ? serviceName
+      ? `${alertTypeName} | ${serviceName}`
+      : alertTypeName
+    : undefined;
+  const tags = ['apm'];
+  if (serviceName) {
+    tags.push(`service.name:${serviceName}`.toLowerCase());
+  }
+  const initialValues = {
+    tags,
+    ...(alertName ? { name: alertName } : {}),
+  };
+
```

Comment:
Setting initial values for alert name and tags via the `initialValues` prop of the `AddAlert` component instead of using `setAlertProperty` in the `ServiceAlertTrigger` component. This allows us to correctly check whether the user changes the alert name/tags and show the confirm on discard modal accordingly. Otherwise the default alert name/tag will always register as a change to the alert